test/distributed/checkpoint/test_traverse.py

```diff
@@ -45,13 +47,13 @@ def collect_data(path, value):
         self.assertEqual(data[("key2", "2")], 3)
 
         self.assertIn(("key3",), data)
-        self.assertEqual(data[("key3",)], torch.tensor([1]))
+        self.assertTrue(torch.equal(data[("key3",)], torch.tensor([1]).to(device)))
```

Comment:
curious why we'd need to change `assertEqual` to `assertTrue(torch.equal(...))`?